test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java

```diff
@@ -288,16 +296,24 @@ protected AggregationContext createAggregationContext(IndexSearcher indexSearche
         );
     }
 
+    /**
+     * {@link ObjectMapper}s to add to the lookup. By default we don't need
+     * any {@link ObjectMapper}s but testing nested objects will require adding some.
+     */
+    protected List<ObjectMapper> objectMappers() {
```

Comment:
This is required by this change because we can no longer mock anything and magic nested fields into place with any object that starts with `nested_`. Now tests that use nested fields need to declare them

Reply:
We couldn't easily mock `MappingLookup` until yesterday because we exposed `FieldTypeLookup`. But now that we don't we could. I don't particularly want to though. I kind of like having a real lookup now. But maybe I've just gotten used to it.

Reply:
Got it, it just seems like this adds some complexity to the aggregation test framework. I don't feel strongly about this though, perhaps the nested/ object mapper set-up here is already a bit complex (as seen in your comment below about `mapperService.hasNested()`).